src/ucp/api/ucp.h

```diff
@@ -331,6 +331,47 @@ enum ucp_ep_close_mode {
 };
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief UCP performance fields and flags
+ *
+ * The enumeration allows specifying which fields in @ref ucp_ep_perf_attr_t are
+ * present and operation flags are used. It is used to enable backward
+ * compatibility support.
+ */
+typedef enum ucp_ep_perf_attr_field {
+    /** Enables @ref ucp_ep_perf_attr_t::message_size */
+    UCP_EP_PERF_ATTR_FIELD_MESSAGE_SIZE       = UCS_BIT(0),
+
+    /** Enables @ref ucp_ep_perf_attr_t::local_memory_type */
+    UCP_EP_PERF_ATTR_FIELD_LOCAL_MEMORY_TYPE  = UCS_BIT(1),
+
+    /** Enables @ref ucp_ep_perf_attr_t::remote_memory_type */
+    UCP_EP_PERF_ATTR_FIELD_REMOTE_MEMORY_TYPE = UCS_BIT(2),
+
+    /** Enables @ref ucp_ep_perf_attr_t::op_type */
+    UCP_EP_PERF_ATTR_FIELD_OP_TYPE            = UCS_BIT(3),
+
+    /** Enables @ref ucp_ep_perf_attr_t::estimated_time */
+    UCP_EP_PERF_ATTR_FIELD_ESTIMATED_TIME     = UCS_BIT(4)
+} ucp_ep_perf_attr_field_t;
+
+
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief All existing UCP operations
+ * 
+ * This enumeration defines all available UCP operations.
+ */
+typedef enum {
+    UCP_OP_TYPE_TAG,    /**< Tag matching */
+    UCP_OP_TYPE_AM,     /**< Active messge */
+    UCP_OP_TYPE_RMA,    /**< Remote memory access */
+    UCP_OP_TYPE_ATOMIC, /**< Atomic operations */
+    UCP_OP_TYPE_STREAM  /**< Stream */
+} ucp_ep_op_type_t;
```

Comment:
Almost identical to https://github.com/openucx/ucx/blob/b72fec8df4ac1ab14e5a3392183215164c7f07cf/src/ucp/api/ucp.h#L140

Reply:
it's not exactly the same, and unfortunately we can't change UCP_FEATURE_xx, but on other hand it would not exactly fit perf query API